src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    const nextState = { ...currentState };
```

Comment:
Instead of creating 2 copies of state, let's keep one and when you are pushing it to the states use spread operator states.push({...currentState})